server/src/main/java/io/deephaven/server/session/SharedTicketResolver.java

```diff
@@ -0,0 +1,288 @@
+            final SessionState.ExportObject<T> source) {
+        if (source.isNonExport()) {
+            throw Exceptions.statusRuntimeException(Code.FAILED_PRECONDITION, "Could not publish '" + logId
+                    + "' to shared ticket '" + toHexString(idForTicket(ticket, logId)) + "' (hex): source must be a"
+                    + " session owned export");
+        }
+        final ByteString sharedId = idForTicket(ticket, logId);
+        final SessionState.ExportObject<?> existing = sharedVariables.putIfAbsent(sharedId, source);
+        if (existing != null) {
+            final String ticketHex = toHexString(sharedId);
+            errorHandler.onError(ExportNotification.State.FAILED, "",
+                    Exceptions.statusRuntimeException(Code.ALREADY_EXISTS, "Could not publish '" + logId
+                            + "' to shared ticket '" + ticketHex + "' (hex): destination already exists"),
+                    null);
+        } else if (onPublish != null) {
+            onPublish.run();
+        }
+    }
+
+    /**
+     * Convenience method to convert from a shared identifier to Flight.Ticket
+     *
+     * @param identifier the shared identifier to convert
+     * @return the flight ticket this descriptor represents
+     */
+    public static Flight.Ticket flightTicketForId(final byte[] identifier) {
+        return Flight.Ticket.newBuilder()
+                .setTicket(ByteString.copyFrom(identifier))
+                .build();
+    }
+
+    /**
+     * Convenience method to convert from a shared identifier to Ticket
+     *
+     * @param identifier the shared identifier to convert
+     * @return the flight ticket this descriptor represents
+     */
+    public static io.deephaven.proto.backplane.grpc.Ticket ticketForId(final byte[] identifier) {
+        return io.deephaven.proto.backplane.grpc.Ticket.newBuilder()
+                .setTicket(ByteString.copyFrom(identifier))
+                .build();
+    }
+
+    /**
+     * Convenience method to convert from a shared identifier to Flight.FlightDescriptor
+     *
+     * @param identifier the shared identifier to convert
+     * @return the flight descriptor this descriptor represents
+     */
+    public static Flight.FlightDescriptor descriptorForId(final byte[] identifier) {
+        return Flight.FlightDescriptor.newBuilder()
+                .setType(Flight.FlightDescriptor.DescriptorType.PATH)
+                .addAllPath(SharedTicketHelper.idToPath(identifier))
+                .build();
+    }
+
+    /**
+     * Convenience method to convert from a Flight.Ticket (as ByteBuffer) to shared identifier
+     *
+     * @param ticket the ticket to convert
+     * @param logId an end-user friendly identification of the ticket should an error occur
+     * @return the query scope name this ticket represents
+     */
+    public static ByteString idForTicket(final ByteBuffer ticket, final String logId) {
```

Comment:
I think a return of `byte[]` would be cleaner? (Or, maybe we could argue for new shared `ByteBuffer` )

Reply:
I need `ByteString` for the map; but I did make these two methods private instead.